dids/src/main/kotlin/web5/sdk/dids/didcore/VerificationMethod.kt

```diff
@@ -0,0 +1,108 @@
+package web5.sdk.dids.didcore
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize
+import com.fasterxml.jackson.databind.annotation.JsonSerialize
+import com.nimbusds.jose.jwk.JWK
+import web5.sdk.dids.JWKSerializer
+import web5.sdk.dids.JwkDeserializer
+
+/**
+ * VerificationMethod expresses verification methods, such as cryptographic
+ * public keys, which can be used to authenticate or authorize interactions
+ * with the DID subject or associated parties.
+ * For example, a cryptographic public key can be used as a verification method with
+ * respect to a digital signature; in such usage, it verifies that the
+ * signer could use the associated cryptographic private key.
+ * Specification Reference: https://www.w3.org/TR/did-core/#verification-methods
+ *
+ * @property id id of the VerificationMethod
+ * @property type references exactly one verification method type. In order to maximize global
+ *  	            interoperability, the verification method type SHOULD be registered in the
+ *                DID Specification Registries: https://www.w3.org/TR/did-spec-registries/
+ * @property controller a value that conforms to the rules in DID Syntax: https://www.w3.org/TR/did-core/#did-syntax
+ * @property publicKeyJwk specification reference: https://www.w3.org/TR/did-core/#dfn-publickeyjwk
+ */
+public class VerificationMethod(
+  public val id: String,
+  public val type: String,
+  public val controller: String,
+  @JsonSerialize(using = JWKSerializer::class)
+  @JsonDeserialize(using = JwkDeserializer::class)
+  public val publicKeyJwk: JWK? = null
+) {
+  /**
+   * Checks type of VerificationMethod.
+   *
+   * @param type The type to check
+   * @return true/false if the type matches
+   */
+  public fun isType(type: String): Boolean {
+    return type == this.type
+  }
+
+  override fun toString(): String {
```

Comment:
@jiyoontbd do you think we should introduce `toJSON()` functions on all of these types? I know we have `dids/src/main/kotlin/web5/sdk/dids/Json.kt` but another idea would be to add the JSON functionality to each DID type, to-and-from JSON strings. idk just a random thought I figured worth considering

Reply:
i think for now i'd like to just have Json.kt i sniped from tbdex-kt to do all the stringify work instead of distributing it to be for each type

Reply:
😎 👍